crates/engine/tree/src/tree/payload_processor/prewarm.rs

```diff
@@ -83,47 +80,25 @@ where
         self.actions_tx.clone()
     }
 
-    /// Spawns the next transactions
-    fn spawn_next(&mut self) {
-        while self.in_progress < self.max_concurrency {
-            if let Some(tx) = self.pending.pop_front() {
-                // increment the in progress counter
-                self.in_progress += 1;
-
-                self.spawn_transaction(tx);
-            } else {
-                break
-            }
+    /// Spawns all pending transactions as blocking tasks by first chunking them.
+    fn spawn_all(&mut self) {
+        let chunk_size = (self.pending.len() / self.max_concurrency).max(1);
```

Comment:
isn't it number of chunks? And the chunk size is basically `max_concurrency`, or am I misunderstanding something? cc @mattsse